analyses/mutational-signatures/04-analyze_de_novo.Rmd

```diff
@@ -0,0 +1,175 @@
+path_to_results <- "results/"
+if (!dir.exists(path_to_results)) {
+  dir.create(path_to_results, recursive = TRUE)
+}
+
+# De novo signatures and exposures list (condensed from sigfit output) file
+de_novo_signatures_file <- file.path(path_to_results, "de_novo_signatures.RDS")
+de_novo_exposures_file <- file.path(path_to_results, "de_novo_exposures.RDS")
+
+# Load cosmic data from sigfit
+data("cosmic_signatures_v3")
+cosmic_names <- row.names(cosmic_signatures_v3)
+```
+
+<br><br>
+First, we need to extract the signatures and exposures from the *de novo* extraction RDS files. We save each into a list `de_novo_signatures` and `de_novo_extractions` separately, and write each list to file. If _both_ files already exist, we can simply read it in the lists.
+
+```{r parse_signatures}
+if (file.exists(de_novo_signatures_file) & file.exists(de_novo_exposures_file))
+{
+  de_novo_signatures <- readr::read_rds(de_novo_signatures_file)
+  de_novo_exposures <- readr::read_rds(de_novo_exposures_file)
+
+} else {
+    
+  # Collect the de novo extracted signatures into a single list
+  # This takes a few minutes to parse.
+  de_novo_signatures <- list()
+  de_novo_exposures <- list()
+  for (this_model in c("poisson" , "multinomial"))
+  {
+    for (this_seed in 1:5){
+  
+      this_name <- glue::glue("seed_{this_seed}_model_{this_model}")
+      filename <- file.path(path_to_input_rds, 
+                            glue::glue("{this_name}.RDS"))
+      #cat(filename)
+      if (file.exists(filename)) {
+        fit <- readr::read_rds(filename)
+        for (k in 2:5){
+          sig_name <- glue::glue("{k}_{this_name}")
+          de_novo_signatures[[ sig_name ]] <- sigfit::retrieve_pars(fit[[k]], par = "signatures") 
+          de_novo_exposures[[ sig_name ]] <- sigfit::retrieve_pars(fit[[k]], par = "exposures") 
+        }
+      }
+    }
+    
+  }
+  # We save this list to a result file
+  readr::write_rds(de_novo_signatures, de_novo_signatures_file)
+  readr::write_rds(de_novo_exposures, de_novo_exposures_file)
+}
+```
+
+<br><br>
+Based on goodness of fit analyses, we expect these *k* to be optimal for the given seed and inference model (noting that we do see strong sensitivity to this starting condition).
+
+```{r gof_k}
+gof <- tibble::tribble(
+  ~model,     ~seed,      ~best_k,
+  #-----------------
+  "poisson",1,3,
+  "multinomial",1,3,
+  "poisson",2,4,
+  "multinomial",2,4,
+  "poisson",3,3,
+  "multinomial",3,3,
+  "poisson",4,3,
+  "multinomial",4,3,  
+  "poisson",5,3,
+  "multinomial",5,3  
+)
+```
+
+
+<br><br>
+Now, we can determine which signatures were extracted and map back to COSMIC names using cosine similarity to determine matching signatures:
+
+```{r determine_sigs}
+
+# Will find the COSMIC equivalent using cosine similarity
+map_match <- function(x) {
+  sort( # Can sort since we just want to see cosmic matches
+    as.numeric( # none of this "optimal assigment ==> 1" business
+      match_signatures(x, cosmic_signatures_v3)
```

Comment:
Information question - `match_signatures()` comes from `sigfit` I assume?

Reply:
Adding a comment to clarify this